diff --git a/src/DraggableMasonryList.tsx b/src/DraggableMasonryList.tsx
index cacb84dbd769c001852547c720696ff2772cbdca..b185d417ebeb3ca1270b6ed3aee320d9e8c8de6b 100644
--- a/src/DraggableMasonryList.tsx
+++ b/src/DraggableMasonryList.tsx
@@ -606,7 +606,18 @@ function DraggableMasonryList<T extends MasonryItemType>({
 
     // --- Drag Handlers ---
     const handleDragStart = useCallback((id: string) => {
-        if (!sortEnabled) return;
+        if (!sortEnabled) {
+            if (onDragStart) {
+                onDragStart({ key: id, fromIndex: -1 });
+            }
+            return;
+        }
+
+        const current = orderedDataRef.current.find(item => keyExtractor(item) === id);
+        const packageName = current?.["packageName"] ?? "";
+        if (packageName?.includes("@empty")) {
+            return;
+        }
 
         setActiveDragId(id);
         isDraggingShared.value = true;
@@ -616,7 +627,7 @@ function DraggableMasonryList<T extends MasonryItemType>({
             layoutWithoutDraggedRef.current = calculateLayout(
                 withoutDragged, numColumns, containerWidth, rowGap, columnGap, keyExtractor
             );
-        }
+        } 
 
         const currentIndex = orderedDataRef.current.findIndex(item => keyExtractor(item) === id);
         dragStartIndexRef.current = currentIndex;
